Remotion/Validation/UnitTests/DefaultValidatorExtensionsTest.cs

```diff
@@ -1,14 +1,416 @@
+    }
+
+    [Test]
+    public void MinLength_ForString_ReturnsLengthValidator ()
+    {
+      var validationMessage = new InvariantValidationMessage("Fake message");
+      var initParameters = new PropertyValidationRuleInitializationParameters(validationMessage);
+      IPropertyValidator createdValidator = null;
+      var addingPropertyValidationRuleBuilder = new Mock<IAddingPropertyValidationRuleBuilder<object, string>>();
+      addingPropertyValidationRuleBuilder.Setup(_ => _.SetValidator(It.IsAny<Func<PropertyValidationRuleInitializationParameters, IPropertyValidator>>()))
+          .Callback<Func<PropertyValidationRuleInitializationParameters, IPropertyValidator>>(func => createdValidator = func.Invoke(initParameters));
+
+      addingPropertyValidationRuleBuilder.Object.MinLength(17);
+
+      addingPropertyValidationRuleBuilder.Verify();
+      Assert.That(createdValidator, Is.InstanceOf<LengthValidator>());
+      var lengthValidator = (LengthValidator)createdValidator;
+      Assert.That(lengthValidator.Min, Is.EqualTo(17));
+      Assert.That(lengthValidator.Max, Is.Null);
+    }
+
+    [Test]
+    public void MaxLength_ForString_ReturnsLengthValidator ()
+    {
+      var validationMessage = new InvariantValidationMessage("Fake message");
+      var initParameters = new PropertyValidationRuleInitializationParameters(validationMessage);
+      IPropertyValidator createdValidator = null;
+      var addingPropertyValidationRuleBuilder = new Mock<IAddingPropertyValidationRuleBuilder<object, string>>();
+      addingPropertyValidationRuleBuilder.Setup(_ => _.SetValidator(It.IsAny<Func<PropertyValidationRuleInitializationParameters, IPropertyValidator>>()))
+          .Callback<Func<PropertyValidationRuleInitializationParameters, IPropertyValidator>>(func => createdValidator = func.Invoke(initParameters));
+
+      addingPropertyValidationRuleBuilder.Object.MaxLength(42);
+
+      addingPropertyValidationRuleBuilder.Verify();
+      Assert.That(createdValidator, Is.InstanceOf<LengthValidator>());
+      var lengthValidator = (LengthValidator)createdValidator;
+      Assert.That(lengthValidator.Min, Is.EqualTo(0));
+      Assert.That(lengthValidator.Max, Is.EqualTo(42));
+    }
+
+    [Test]
+    public void ExactLength_ForString_ReturnsLengthValidator ()
+    {
+      var validationMessage = new InvariantValidationMessage("Fake message");
+      var initParameters = new PropertyValidationRuleInitializationParameters(validationMessage);
+      IPropertyValidator createdValidator = null;
+      var addingPropertyValidationRuleBuilder = new Mock<IAddingPropertyValidationRuleBuilder<object, string>>();
+      addingPropertyValidationRuleBuilder.Setup(_ => _.SetValidator(It.IsAny<Func<PropertyValidationRuleInitializationParameters, IPropertyValidator>>()))
+          .Callback<Func<PropertyValidationRuleInitializationParameters, IPropertyValidator>>(func => createdValidator = func.Invoke(initParameters));
```

Comment:
Layout: we're trying to do those multiline statements for mock setups with one call per line. This helps with reading the setup syntax by giving the eye a single line to read.
```suggestion
      addingPropertyValidationRuleBuilder
          .Setup(_ => _.SetValidator(It.IsAny<Func<PropertyValidationRuleInitializationParameters, IPropertyValidator>>()))
          .Callback<Func<PropertyValidationRuleInitializationParameters, IPropertyValidator>>(func => createdValidator = func.Invoke(initParameters));
```